### DISCUSSION TITLE
How to properly perform univariate forecasting without covariate in sktime.forecasting.pytorchforecasting

### DISCUSSION THREAD

**Discussion by jobs-git** [alan-turing-institute/sktime#8260] [Q&A] [answered]
I used the API example from sktime site, but unfortunately this seems to be adopted for future covariate where its forecasting can be done given a past series and its future covariate, see:

https://www.sktime.net/en/latest/api_reference/auto_generated/sktime.forecasting.pytorchforecasting.PytorchForecastingNHiTS.html

So I revised to code to make sure its univariate, I set the X values to be equal to the label values. Then to perform forecasting in without future covariate, I limited the training points for X to past data sets only as shown in the code below. Unfortunately, its not producing outputs that is within the magnitude of the data set. However, the prediction returns empty.

So how to properly perform univariate forecasting without covariate in sktime.forecasting.pytorchforecasting?

```python
from sktime.forecasting.base import ForecastingHorizon
from sktime.forecasting.pytorchforecasting import PytorchForecastingNHiTS
from sktime.utils._testing.hierarchical import _make_hierarchical
from sklearn.model_selection import train_test_split

data = _make_hierarchical(
    hierarchy_levels=(1, 500), max_timepoints=50, min_timepoints=50, n_columns=3
)

# define forecast horizon
max_prediction_length = 5
fh = ForecastingHorizon(range(1, max_prediction_length + 1), is_relative=True)

# make uni variate only
data["c0"] = data["c2"].values

x = data[["c0"]]
y = data["c2"].to_frame()
X_train, X_test, y_train, y_test = train_test_split(
    x, y, test_size=0.2, train_size=0.8, shuffle=False
)

# limit X data to past data sets only
X_train = X_train.groupby(level=list(range(len_levels - 1))).apply(
    lambda x: x.droplevel(list(range(len_levels - 1))).iloc[:-max_prediction_length]
)

X_test = X_test.groupby(level=list(range(len_levels - 1))).apply(
    lambda x: x.droplevel(list(range(len_levels - 1))).iloc[:-max_prediction_length]
)

len_levels = len(y_test.index.names)

y_test = y_test.groupby(level=list(range(len_levels - 1))).apply(
    lambda x: x.droplevel(list(range(len_levels - 1))).iloc[:-max_prediction_length]
)

# define the model
model = PytorchForecastingNHiTS(
    trainer_params={
        "max_epochs": 50,  # for quick test
        "limit_train_batches": 10,  # for quick test
    },
)

# fit and predict
model.fit(y=y_train, X=X_train, fh=fh) # doctest skip
y_pred = model.predict(fh, X=X_test, y=y_test)
print (y_pred.to_string())
```

**Comment by jobs-git** [comment, marked as answer]
Just use y="univariate data"
